lib/features/composer/presentation/controller/rich_text_controller.dart

```diff
@@ -0,0 +1,42 @@
+
+import 'package:core/utils/app_logger.dart';
+import 'package:get/get.dart';
+import 'package:html_editor_enhanced/html_editor.dart';
+import 'package:model/mailbox/select_mode.dart';
+import 'package:tmail_ui_user/features/composer/presentation/model/rich_text_option.dart';
+
+class RichTextController extends GetxController {
+
+  final formattingOptionsState = SelectMode.INACTIVE.obs;
+
+  final listRichTextOptionSupported = [
+      RichTextOption.bold(),
+      RichTextOption.italic(),
+      RichTextOption.underline()
+  ].obs;
+
+  void toggleRichTextButton() {
+    final currentState = formattingOptionsState.value;
+    final newState = currentState == SelectMode.ACTIVE
+        ? SelectMode.INACTIVE
+        : SelectMode.ACTIVE;
+    formattingOptionsState.value = newState;
+  }
+
+  bool get isFormattingOptionsEnabled =>
+      formattingOptionsState.value == SelectMode.ACTIVE;
```

Comment:
IMO, you should add another `selectMode` for composer in separate with other selectMode

Reply:
I think it is not so important here. SelectMode represents a state object. It's too obvious when setting an object's state.

Reply:
because when we have other change for the list or the composer, I don't want it impact to other

Reply:
What are other changes? What options do you have for it? Please clarify what you mean here?

Reply:
Ok. I created a separate state object for it
```dart
enum FormattingOptionsState {
   enable,
   disable
}
```

Reply:
According to the design we don't need this object